src/expr/type_node.cpp

```diff
@@ -201,6 +202,32 @@ CardinalityClass TypeNode::getCardinalityClass()
   return ret;
 }
 
+bool TypeNode::isCardinalityLessThan(size_t n)
+{
+  if (isBoolean())
+  {
+    return n > 2;
+  }
+  if (isBitVector())
+  {
+    return std::log2(n) > getBitVectorSize();
+  }
+  if (isFloatingPoint())
+  {
+    return std::log2(n) > (getFloatingPointExponentSize()
```

Comment:
I think this can be made more precise, see: https://github.com/cvc5/cvc5/blob/main/src/theory/fp/theory_fp_type_rules.cpp#L764.

Can you make that computation a separate utility, e.g. `Integer fp::getFloatingPointCardinality(TypeNode tn)` then call this method from here?

Reply:
Why only for FP and not also for BV?

Reply:
IMO, BV is simple enough to be inlined and optimized via the direct log computation.

Reply:
FP is as simple as BV, IMO.

Reply:
I do not agree that for FP a utility is necessary, it's the same as for BV.

Reply:
Note that I would be in favor to introduce such a utility if we need it more than once, but then, for consistency, it should be introduced for BV and FP.

Reply:
The cardinality of FP is based on this computation: https://github.com/cvc5/cvc5/blob/main/src/theory/fp/theory_fp_type_rules.cpp#L764
```
  /*
   * 1                    NaN
   * 2*1                  Infinities
   * 2*1                  Zeros
   * 2*2^(s-1)            Subnormal
   * 2*((2^e)-2)*2^(s-1)  Normal
   *
   *  = 1 + 2*2 + 2*((2^e)-1)*2^(s-1)
   *  =       5 + ((2^e)-1)*2^s
   */

Reply:
Right now this utility would be needed in `fp::CardinalityComputer::computeCardinality` and in this method.  The former could be deleted eventually but I'd prefer the method be in the fp namespace.

Reply:
Yes, the NaN case is tricky and that's actually a good point. NaN is not a single value, but a set of values, however NaN != NaN (always, no matter if the NaN values are equal), so in this particular case, we should consider it as a single value.

Reply:
Then we are back to big num computation for FP, though. And I can just use `computeCardinality` of the type rules....